### PR TITLE
[triplets][wasm32-emscripten] Find emscripten installation based on PATH

### PR DESCRIPTION
**Pull request by sbc100**
Not all emscripten users are emsdk users so `$EMSDK` won't always be
present.  However its resonable to assume that all emscripten users
(including emsdk users) will have `emcc` (the esmcripten compiler) in
their `$PATH`

Keep the old method of locating emscripten (which is fragile since it
depends on the internal layout of emsdk) around in case there are any
users that set `$EMSDK` themselves without adding emcc to the `$PATH`.
This is unlikely since emsdk, when used in the normal way, takes care
of adding the emscripten directory to `$PATH`.

### CODE REVIEW

**Comment by sbc100** [comment]
@MoAlyousef

File: triplets/community/wasm32-emscripten.cmake [hunk 1]
```diff
@@ -1,15 +1,20 @@
 set(VCPKG_ENV_PASSTHROUGH EMSDK PATH)
 
-if(NOT DEFINED ENV{EMSDK})
-   message(FATAL_ERROR "The EMSDK environment variable must be defined")
+find_path(EMSCRIPTEN_ROOT "emcc")
+if(NOT EMSCRIPTEN_ROOT)
+    # Old-lookup method, try to infer emscripten direcotyr based on $EMSDK environment variable
```
Comment (traversaro) [inline triplets/community/wasm32-emscripten.cmake:5]:
```suggestion
    # Old-lookup method, try to infer emscripten directory based on $EMSDK environment variable
```

**Comment by MoAlyousef** [comment]
This, along woth the fallback, looks more robust. I recall at some point Emscripten changed the internal directory layout, that was before adding this community triplet, so depending on the layout might work today but break in a future release of Emscripten.

**Comment by sbc100** [comment]
> This, along woth the fallback, looks more robust. I recall at some point Emscripten changed the internal directory layout, that was before adding this community triplet, so depending on the layout might work today but break in a future release of Emscripten.

Indeed, and also importantly not all emscripten users use emsdk (in particular those who work on emscripten itself, such as myself).

**Review by JonLiu1993 — APPROVED** [review]

File: triplets/community/wasm32-emscripten.cmake [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
 
-if(NOT DEFINED ENV{EMSDK})
-   message(FATAL_ERROR "The EMSDK environment variable must be defined")
+find_path(EMSCRIPTEN_ROOT "emcc")
+if(NOT EMSCRIPTEN_ROOT)
+    # Old-lookup method, try to infer emscripten directory based on $EMSDK environment variable
+    if(NOT DEFINED ENV{EMSDK})
+        message(FATAL_ERROR "emcc compiler not found in PATH")
+    endif()
+    set(EMSCRIPTEN_ROOT "ENV{EMSDK}//upstream/emscripten")
```
Comment (ras0219-msft) [inline triplets/community/wasm32-emscripten.cmake:9]:
```suggestion
    set(EMSCRIPTEN_ROOT "$ENV{EMSDK}/upstream/emscripten")
```

**Review by ras0219-msft — COMMENTED** [review]
LGTM after my two suggestions. Thanks for working on Emscripten -- it's _awesome_!

**Comment by sbc100** [comment]
> [CLA assistant check](https://cla.opensource.microsoft.com/microsoft/vcpkg?pullRequest=16534)
> Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/vcpkg?pullRequest=16534) before we can accept your contribution.
> 
>  sbc100 [sign now](https://cla.opensource.microsoft.com/microsoft/vcpkg?pullRequest=16534)
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/microsoft/vcpkg?pullRequest=16534) it.

This is odd.. my first commit didn't seem to have an issue: #16522.    Also the first upload on this PR didn't have an issue either.   Is this because used the github "apply suggestion" feature and so the author/commiter is somehow different

**Comment by vicroms** [comment]
>Is this because used the github "apply suggestion" feature and so the author/commiter is somehow different

That may be the case, @ras0219-msft any insight into how CLA works? 
Right now we're blocked from merging this PR because of it.

**Review by vicroms — APPROVED** [review]

**Comment by sbc100** [comment]
Abandoning this for now until I figure out CLA issues with my employer.

**Pull request closed without merging**